frontend/packages/dev-console/src/components/edit-deployment/CreateDeploymentPage.tsx

```diff
@@ -0,0 +1,40 @@
+import * as React from 'react';
+import Helmet from 'react-helmet';
+import { useTranslation } from 'react-i18next';
+import { match as Rmatch } from 'react-router-dom';
+import { DeploymentModel } from '@console/internal/models';
+import { DeploymentKind } from '@console/internal/module/k8s';
+import EditDeployment from './EditDeployment';
+import { getDefaultDeployment } from './utils/create-deployment-utils';
+
+export type CreateDeploymentPageProps = {
+  match: Rmatch<{
+    ns: string;
+    url: string;
+  }>;
+};
+
+const CreateDeploymentPage: React.FC<CreateDeploymentPageProps> = ({ match }) => {
+  const { t } = useTranslation();
+  const {
+    params: { ns: namespace },
+    url,
+  } = match;
+
+  const createResourceType = url.split('/')[4] === DeploymentModel.plural && DeploymentModel.label;
```

Comment:
@Lucifergene I don't think we need this check here as we know this form will open when the URL is `/k8s/ns/:ns/deployments/~new/form`. Right?

Reply:
```suggestion
  const createResourceType = DeploymentModel.label;
```